deploy/hs-delete-custom-cmo-config/10-hs-delete-custom-cmo.Job.yaml

```diff
@@ -0,0 +1,23 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: hs-delete-custom-cmo-config
+  namespace: openshift-delete-custom-cmo-config
+spec:
+  template:
+    spec:
+      containers:
+      - name: hs-delete-custom-cmo-config
+        image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+        command:
+        - bash
+        - -c
+        - |
+          readarray -t namespaces < <(oc get cd -l ext-hypershift.openshift.io/cluster-type=management-cluster -A -o go-template='{{range .items}}{{.metadata.namespace}}{{"\n"}}{{end}}');
+          for ns in "${namespaces[@]}"; do
+            oc patch syncset ext-cluster-monitoring-operator-config -n "${ns}" --type merge -p '{"spec":{"resourceApplyMode":"Upsert"}}';
```

Comment:
I may miss some context here, can you please elaborate where these syncsets come from if they're not managed by managed-cluster-config and why we need to patch them like this?

Reply:
> where these syncsets come from if they're not managed by managed-cluster-config

These SyncSets are created by Fleet Manager, [templates](https://gitlab.cee.redhat.com/service/osd-fleet-manager/-/blob/main/config/resources/managed-cluster-cmo-config.yaml) are converted to SS by FM. We'll be deprecating FM-managed CMO config so the SS won't be needed anymore, ref: https://gitlab.cee.redhat.com/service/osd-fleet-manager/-/merge_requests/475

> why we need to patch them like this?

The SynSets is patched to `Upsert` so that the on-cluster config map is not deleted if this SyncSet on Hive is deleted. This ensures that the monitoring CO isn't degraded until the stock/default CMO config (https://github.com/openshift/managed-cluster-config/pull/1663) is deployed to all MCs.